community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/ast/LikeParser.scala

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compiler.v2_2.ast
+
+import org.parboiled.scala._
+
+//TODO should option be supported? In standard?
+/**
+ * Parser that parses a Like expression into [[ParsedLike]] which contains tokens of [[LikeOp]].
+ *
+ * A Like expression can contain the special characters:
+ *  - % wildcard
+ *  - _ matches exactly one character
+ *  - [XY] optionally match any of the characters
+ */
+case object LikeParser extends Parser {
```

Comment:
I would assume with a full regex parser this would be a problem, but with only '%' and '_' wildcards, there does not seem to be any way the internal parsing can generate an error (other than a bug in our code, of course). So is it not sufficient to return the offset of the beginning of the pattern string on any inner parser errors?